net/net.go

```diff
@@ -175,3 +184,131 @@ func ListenOnUDPRandomPort(ip string) (*net.UDPConn, error) {
 
 	return net.ListenUDP("udp4", &localAddr)
 }
+
+// MatchIP is used to determine whether @pattern and @host:@port match, It's supports subnet/range
+func MatchIP(pattern, host, port string) bool {
```

Comment:
the param `port` seems redundant for the func，it check whether ip is match

Reply:
When pattern has port, it will be check.